### PR TITLE
test: Port tests in `partitioned_csv.rs` to sqllogictest

### LINKED ISSUE #8208: Port tests in `partitioned_csv.rs` to sqllogictest
### Is your feature request related to a problem or challenge?

Part of https://github.com/apache/arrow-datafusion/issues/6195. We are trying to
1. keep the DataFusion code base clean and easy to contribute to and modify
2. Understand what features are covered and what are not

Part of doing so is having a single location for most test coverage so new tests can be added easily and people can follow the existing patterns easily

### Describe the solution you'd like

Port the tests in https://github.com/apache/arrow-datafusion/blob/main/datafusion/core/tests/sql/partitioned_csv.rs to
`partitioned_csv.slt` in https://github.com/apache/arrow-datafusion/tree/main/datafusion/sqllogictest/test_files

Note you can create catalogs and schema via SQL now

Catalogs: https://arrow.apache.org/datafusion/user-guide/sql/ddl.html#create-database
Schema: https://arrow.apache.org/datafusion/user-guide/sql/ddl.html#create-schema

Notes:
1. Instructions for running sqllogitest are here: https://github.com/apache/arrow-datafusion/tree/main/datafusion/sqllogictest

### PR DESCRIPTION
**Pull request by simicd**
## Which issue does this PR close?
Closes #8208

## Rationale for this change
Migrate tests from Rust to sqllogictest

## What changes are included in this PR?
- Remove duplicated code from `partitioned_csv.rs`
- Create new unit test with partitioned csv files in `csv_files.slt`

## Are these changes tested?
Yes

## Are there any user-facing changes?
No

### CODE REVIEW

File: datafusion/core/tests/sql/select.rs [hunk 1]
```diff
@@ -482,7 +482,7 @@ async fn sort_on_window_null_string() -> Result<()> {
 async fn test_prepare_statement() -> Result<()> {
     let tmp_dir = TempDir::new()?;
     let partition_count = 4;
-    let ctx = partitioned_csv::create_ctx(&tmp_dir, partition_count).await?;
+    let ctx = create_ctx_with_partition(&tmp_dir, partition_count).await?;
```
Comment (simicd) [inline datafusion/core/tests/sql/select.rs:485]: The same code existed in `partitioned_csv.rs` and `mod.rs` - dropped the former and replaced with function calls to `mod.rs`

File: datafusion/sqllogictest/test_files/csv_files.slt [hunk 2]
```diff
@@ -63,3 +63,76 @@ id6 value"6
 id7 value"7
 id8 value"8
 id9 value"9
+
+
+# Read partitioned csv
```
Comment (simicd) [inline datafusion/sqllogictest/test_files/csv_files.slt:68]: `partitioned_csv.rs` did not have any unit tests so I created a new one to test reading from partitioned csv files.

File: datafusion/core/tests/sql/partitioned_csv.rs [hunk 3]
```diff
@@ -1,77 +0,0 @@
...
-
-use arrow::datatypes::{DataType, Field, Schema, SchemaRef};
-use datafusion::{
-    error::Result,
-    prelude::{CsvReadOptions, SessionConfig, SessionContext},
-};
-use tempfile::TempDir;
-
-/// Generate CSV partitions within the supplied directory
-fn populate_csv_partitions(
```
Comment (simicd) [inline datafusion/core/tests/sql/partitioned_csv.rs:30]: The exact same setup code exists in `mod.rs` so deleted this one here.

https://github.com/apache/arrow-datafusion/blob/ae0f401d89cf9dc8b717e6b95f73ed4f3be9798b/datafusion/core/tests/sql/mod.rs#L234-L282

**Review by alamb — APPROVED** [review]
Thank you @simicd  -- this looks great

**Pull request merged**